src/main/java/domain/Board.java

```diff
@@ -0,0 +1,178 @@
+package domain;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class Board {
+
+    private final Map<BoardPosition, Piece> pieces;
+
+    public Board(final Map<BoardPosition, Piece> pieces) {
+        validateNotNull(pieces);
+        this.pieces = new HashMap<>(pieces);
+    }
+
+    private void validateNotNull(final Map<BoardPosition, Piece> pieces) {
+        if (pieces == null) {
+            throw new IllegalArgumentException("보드는 기물들을 가져야합니다.");
+        }
+    }
+
+    public static Board initialize() {
+        final Map<BoardPosition, Piece> pieces = createInitializePieces();
+
+        return new Board(pieces);
+    }
+
+    private static Map<BoardPosition, Piece> createInitializePieces() {
+        final Map<BoardPosition, Piece> pieces = new HashMap<>();
+
+        PieceType.CANNON.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.CANNON, team))));
+        PieceType.CHARIOT.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.CHARIOT, team))));
+        PieceType.ELEPHANT.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.ELEPHANT, team))));
+        PieceType.GENERAL.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.GENERAL, team))));
+        PieceType.GUARD.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.GUARD, team))));
+        PieceType.HORSE.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.HORSE, team))));
+        PieceType.쭈.getInitialPosition()
+            .forEach((team, positions) -> positions
+                .forEach(position -> pieces.put(position, new Piece(PieceType.쭈, team))));
```

Comment:
```suggestion
   Arrays.stream(PieceType.values())
        .forEach(type -> type.getInitialPosition()
            .forEach((team, positions) -> positions
                .forEach(position -> pieces.put(position, new Piece(type, team)))));
```

PieceType.values()을 사용하여 중복코드를 제거할 수 있을 것 같아요.

Reply:
넵! 이견 없습니다!

처음에 해당 방식을 도입할 때는, 명시적으로 데이터를 넣어주는 방식보다는 람다식을 활용해 함수형 스타일로 구성하고 싶었습니다. 마치 SQL에서 초기 데이터를 주입하듯이 말이죠 😄

다만, 다시 생각해보니 라라가 추천해주신 방식도 충분히 제 의도를 잘 드러내면서, 동시에 가독성 측면에서도 더 나은 선택일 수 있다고 느꼈고 ,[f3faaa7](https://github.com/woowacourse/java-janggi/pull/59/commits/f3faaa787d553609d23ab25ba48af6dcf9cdb6a8)에 반영했습니다! 감사합니다!! 😎